storagemarket/impl/client.go

```diff
@@ -204,6 +205,10 @@ func (c *Client) ProposeStorageDeal(
 		return nil, xerrors.Errorf("computing commP failed: %w", err)
 	}
 
+	if uint64(pieceSize.Padded()) > info.SectorSize {
```

Comment:
Great idea to do this check early -- even if the provider would reject the deal eventually we should go ahead and error if we know the deal won't be accepted